src/Mod/Fem/femsolver/calculix/writer.py

```diff
@@ -909,6 +912,49 @@ def write_constraints_selfweight(self, f):
         # different element sets for different density
         # are written in the material element sets already
 
+    # ********************************************************************************************
+    # constraints centrif
+    def write_constraints_centrif(self, f):
+        if not self.centrif_objects:
+            return
+
+        # write constraint to file
+        f.write("\n***********************************************************\n")
+        f.write("** Centrif Constraint\n")
+        f.write("** written by {} function\n".format(sys._getframe().f_code.co_name))
+        for femobj in self.centrif_objects:
+            # femobj --> dict, FreeCAD document object is femobj["Object"]
+            centrif_obj = femobj["Object"]
+            refobj=centrif_obj.AxisReference[0][0]
+            subobj=centrif_obj.AxisReference[0][1][0]
+            axis=refobj.Shape.getElement(subobj)
+
+            if  axis.Curve.TypeId == "Part::GeomLine":
+                axiscopy=axis.copy()        # apply global placement to copy
+                axiscopy.Placement = refobj.getGlobalPlacement()
+                direction = axiscopy.Curve.Direction
+                location = axiscopy.Curve.Location
+            else:             # no line found, set default
+                location = FreeCAD.Vector(0., 0., 0.)
+                direction = FreeCAD.Vector(0., 0., 1.)
+
+            f.write("** " + centrif_obj.Label + "\n")
+            f.write("*DLOAD\n")
+            f.write(
+                "{},CENTRIF,{:.13G},{:.13G},{:.13G},{:.13G},{:.13G},{:.13G},{:.13G}\n"
```

Comment:
This is an example where I don't understand why the code is as it is.

Reply:
ccx uses F20.0 FORTRAN input fields, please see in dload.f in ccx's source.
A discussion can be found here [https://forum.freecadweb.org/viewtopic.php?f=18&t=22759&#p176578](url) 
So please try "{:.13G}".format(math.sqrt(2.)*-1e100) and count char yourself.
I will add a comment

Reply:
This is what I mean. Another developer without this special knowledge should be able to understand the code. For example for me it was not clear that you write FORTRAN code etc.

But also the pure internal python code should get some more comments. When I started with FC I had a lot of troubles to understand the code. It was doable but costs much more time than having here and there some comments.

Reply:
link in the regard of fortran is wrong: https://forum.freecadweb.org/viewtopic.php?f=18&t=22759&#p176578

Reply:
https://github.com/berndhahnebach/FreeCAD_bhb/commit/00af280c239